File: week_practice/exam_3_practice/BullsNCows/BullsNCows/GameLogic.cpp
#include "stdafx.h"
#include "GameLogic.h"

GameLogic::GameLogic() {
  game = new NumberGenerator;
  game->generateNumber();
}

GameLogic::~GameLogic() {
  delete game;
}

void GameLogic::setUserNumber(std::string input) {
  usernumber = input;
  return;
}

std::string GameLogic::getUserNumber() {
  return usernumber;
}

bool GameLogic::isBull(char digit, int place) {
  for (int i = 0; i < game->getGeneratedNumber().length(); i++) {
    if (digit == game->getGeneratedNumber()[i] && i == place) {
      return true;
    }
  }
  return false;
}

bool GameLogic::isCow(char digit, int place) {
  for (int i = 0; i < game->getGeneratedNumber().length(); i++) {
    if (digit == game->getGeneratedNumber()[i] && i != place) {
      return true;
    }
  }
  return false;
}

bool GameLogic::isWon() {
  return countBulls() == 4;
}

int GameLogic::countBulls() {
  int counter = 0;
  for (int i = 0; i < usernumber.length(); i++) {
    if (isBull(usernumber[i], i)) {
      counter++;
    }
  }
  return counter;
}

int GameLogic::countCows() {
  int counter = 0;
  for (int i = 0; i < usernumber.length(); i++) {
    if (isCow(usernumber[i], i)) {
      counter++;
    }
  }
  return counter;
}

File: week_practice/csharp_practice/BullsAndCows/BullsAndCows/GameLogic.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BullsAndCows
{
	class GameLogic
	{
		private NumberGenerator randomnumber;
		private string _userguess;

		public GameLogic()
		{
			randomnumber = new NumberGenerator();
			randomnumber.generateNumber();
		}
		public string userguess
		{
			get { return _userguess; }
			set { _userguess = value; }
		}
		public bool isCow(char digit, int index)
		{
			for(int i = 0; i < randomnumber.getGeneratedNumber.Length; i++)
			{
				if (digit == randomnumber.getGeneratedNumber[i] && index != i)
				{
					return true;
				}
			}
			return false;
		}
		public bool isBull(char digit, int index)
		{
			for (int i = 0; i < randomnumber.getGeneratedNumber.Length; i++)
			{
				if (digit == randomnumber.getGeneratedNumber[i] && index == i)
				{
					return true;
				}
			}
			return false;
		}
		public bool isWon()
		{
			return randomnumber.getGeneratedNumber == _userguess;
		}
		public int countBulls()
		{
			int counter = 0;
			int index = 0;
			foreach(char c in _userguess)
			{
				if (isBull(c, index))
				{
					counter++;
				}
				index++;
			}
			return counter;
		}
		public int countCows()
		{
			int counter = 0;
			int index = 0;
			foreach (char c in _userguess)
			{
				if (isCow(c, index))
				{
					counter++;
				}
				index++;
			}
			return counter;
		}
	}
}
